File: LoginActivity/app/src/main/java/com/example/loginactivity/MainActivity.java
package com.example.loginactivity;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.util.Consumer;
import com.example.loginactivity.Retrofit.IMyService;
import com.example.loginactivity.Retrofit.RetrofitClient;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.schedulers.Schedulers;
import retrofit2.Retrofit;

public class MainActivity extends AppCompatActivity {

    TextView create_account;
    EditText email, password;
    Button login;

    CompositeDisposable compositeDisposable = new CompositeDisposable();
    IMyService iMyService;

    @Override
    protected void onStop() {
        compositeDisposable.clear();
        super.onStop();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Init services
        Retrofit retrofitClient = RetrofitClient.getInstance();
        iMyService = retrofitClient.create(IMyService.class);

        //Init view
        email = (EditText) findViewById(R.id.email);
        password = (EditText) findViewById(R.id.password);
        login = (Button) findViewById(R.id.login);
        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loginUser(email.getText().toString(),
                        password.getText().toString());
            }
        });
        create_account = (TextView) findViewById(R.id.create_account);
        create_account.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                View register_layout = LayoutInflater.from(MainActivity.this)
                        .inflate(R.layout.register_layout,null);
            }
        });

    }

    private void loginUser(String email, String password) {
        if (TextUtils.isEmpty(email)) {
            Toast.makeText(this,"Email cannot be empty ", Toast.LENGTH_SHORT).show();
            return;
        }
        if (TextUtils.isEmpty(password)) {
            Toast.makeText(this,"Password cannot be empty ", Toast.LENGTH_SHORT).show();
            return;
        }

        compositeDisposable.add(iMyService.loginUser(email, password)
        .subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(new Consumer<String>(){
            @Override
            public void accept(String response) throws Exception{
                Toast.makeText(MainActivity.this,""+response, Toast.LENGTH_SHORT).show();
            }

        }));
    }
}

File: DICEROLLER/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private fun rollDice() {
        // Create new Dice object with 6 sides and roll it
        val dice = DICE(6)
        val diceroll = dice.roll()
        val resultTextView: TextView = findViewById(R.id.textView)
        resultTextView.text = diceroll.toString()
        val diceImage: ImageView = findViewById(R.id.imageView)
        val drawableResource = when (diceroll) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }

        diceImage.setImageResource(drawableResource)

        val luckyNumber = 6
/**
        if (diceroll == luckyNumber) {
            println("You win!")
        } else if (diceroll == 1) {
            println("So sorry! You rolled a 1. Try again!")
        } else if (diceroll == 2) {
            println("Sadly, you rolled a 2. Try again!")
        } else if (diceroll == 3) {
            println("Unfortunately, you rolled a 3. Try again!")
        } else if (diceroll == 4) {
            println("No luck! You rolled a 4. Try again!")
        } else {
            println("Apologies! you rolled a 5. Try again!")
        }**/
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val RollButton: Button = findViewById(R.id.button)
        RollButton.setOnClickListener {
            Toast.makeText(this, "Dice Rolled!", Toast.LENGTH_SHORT).show()
            rollDice()
        }
    }
}
class DICE(private val numside: Int) {
    fun roll(): Int {
        return (1..numside).random()
    }
}